src/Chapter2.hs

```diff
@@ -727,7 +747,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concatMap (\x -> replicate x x) l 
```

Comment:
Great usage of the standard function in here 🏈 
You can also apply eta-reduction in here 😉